doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -738,6 +738,101 @@ When a focused goal is proved, Rocq displays a message suggesting use of
    - "None": this makes bullets inactive.
    - "Strict Subproofs": this makes bullets active (this is the default behavior).
 
+.. _named_goals:
+
+Named goals
+~~~~~~~~~~~
+
+You can focus on a goal by using its name. Goals do not have a name by default,
+but a name can be given by using :n:`refine ?[@ident]`, or generated using the
+:flag:`Generate Goal Names` flag.
+
+.. flag:: Generate Goal Names
+
+   Enables automatic generation of goal names for the :tacn:`induction`,
+   :tacn:`destruct` and :tacn:`eapply` tactics. For :tacn:`induction` and
+   :tacn:`destruct`, the subgoal takes the name of the corresponding
+   constructor. For :tacn:`eapply`, the subgoal takes the name of the
+   corresponding hypothesis.
+
+   This option makes it possible to write proofs with multiple subgoals that do
+   not depend on the order in which constructors were defined, but instead rely
+   on the constructor names. If you use bullets or numbers, reordering
+   constructors will break the proof.
+
+   For proofs that use nested :tacn:`induction` or case analysis, qualified
+   names such as `true.false` are used to disambiguate subgoals (see an example
+   :ref:`here <qualified-goal-names>`).
+
+   .. example:: Automatic generation of goal names
+
+      For the example given :ref:`here <example-working-with-named-goals>`,
+      names are generated for both the base case and the induction case.
+
+      .. rocqtop:: in
+
+         Set Generate Goal Names.
+         Goal forall n, n + 0 = n.
+
+      .. rocqtop:: all
+
+         induction n.
+         [O]: { (* O and S are the constructors for nat. *)
+           reflexivity.
+
+      .. rocqtop:: in abort
+
+         }
+
+      If a goal comes from a variable that failed to instantiate (e.g. when
+      using :tacn:`eapply` or other `e*` tactics), the goal is named after the
+      variable:
```

Comment:
should use the term "existential variable" here.  "failed to instantiate" is needlessly vague.  Mention that `?Rwf` and `?R`come from the definition of `well_founded_ind`.